### PR TITLE
Executor: Dispersal service mempool adapter

### PR DESCRIPTION
**Pull request by bacv**
## 1. What does this PR implement?

Executor dispersal mempool adapter implementation.

## Checklist

* [x] 1. Description added.
* [x] 2. Context and links to Specification document(s) added.
* [x] 3. Main contact(s) (developers and specification authors) added
* [x] 4. Implementation and Specification are 100% in sync including changes. This is critical.
* [x] 5. Link PR to a specific milestone.

### CODE REVIEW

File: nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+use nomos_core::da::blob::info::DispersedBlobInfo;
+use nomos_core::header::HeaderId;
+use nomos_da_sampling::backend::DaSamplingServiceBackend;
+use nomos_mempool::{
+    backend::MemPool, network::NetworkAdapter as MempoolAdapter, DaMempoolService, MempoolMsg,
+};
+use overwatch_rs::{
+    services::{relay::OutboundRelay, ServiceData},
+    DynError,
+};
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs:15]: nitpick: this ones would be internal (although they are crates they are internal to us).

File: nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    for KzgrsMempoolAdapter<
+        DaPoolAdapter,
+        DaPool,
+        SamplingBackend,
+        SamplingNetworkAdapter,
+        SamplingRng,
+        SamplingStorage,
+    >
+where
+    DaPool: MemPool<BlockId = HeaderId, Key = [u8; 32]>,
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs:64]: that `[u8; 32]` is probably telling us that we need at least a type alias.
BTW, Why do we have a `BlockId` in the da mempool? I forgot, but may it be a leftover from the certificate?

File: nomos-services/data-availability/dispersal/src/backend/kzgrs.rs [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
+// crates
 use futures::StreamExt;
 use itertools::izip;
 use kzgrs_backend::common::blob::DaBlob;
 use kzgrs_backend::common::{build_blob_id, Column, ColumnIndex};
-use kzgrs_backend::encoder;
 use kzgrs_backend::encoder::EncodedData;
+use kzgrs_backend::{dispersal, encoder};
 use nomos_core::da::{BlobId, DaDispersal, DaEncoder};
 use overwatch_rs::DynError;
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/dispersal/src/backend/kzgrs.rs:12]: nitpick: also internal (?)

**Review by danielSanchezQ — CHANGES_REQUESTED** [review]
Overall its what needs to be. Lets fix the little stuff and get it merged. Thanks!

File: nomos-services/data-availability/dispersal/src/backend/kzgrs.rs (hunk 3 above)
Comment (bacv) [reply nomos-services/data-availability/dispersal/src/backend/kzgrs.rs:12]: Up until now I considered internal modules to be `crate` and `super`. I'll fix it, just want to confirm.

File: nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs (hunk 2 above)
Comment (bacv) [reply nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs:64]: Agree on type alias, in this case it needs to be from Kzgrs backend (blobId), will add if not existing already. BlockId is not related to certificate afaik, will double check.

File: nomos-services/data-availability/dispersal/src/backend/kzgrs.rs (hunk 3 above)
Comment (danielSanchezQ) [reply nomos-services/data-availability/dispersal/src/backend/kzgrs.rs:12]: yeah, maybe I should have clarified on that. internal would be anything we own.

File: nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs (hunk 2 above)
Comment (bacv) [reply nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs]: `BlockId` is used for marking in block and other operations related to mempool backend.

**Review by danielSanchezQ — APPROVED** [review]
:rocket:

**Pull request merged**